sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -419,66 +513,98 @@ struct tpu_params<tpu::amx, Ta, Tb, Tc, 0, 0, 0,
+```c++
+// User can provide sizes besides the types and tpu_params can assert
+  if they are supported or not
+// in this case, an assertion will happens as 16 is not a supported size for M
+using myparams = tpu_params<tpu::xmx16, int8_t, int8_t, int, 16, 16, 32>;
+size_t NDRangeM = M / myparams::M;  //Assertion would happen at this line
+size_t NDRangeN = N / myparams::N;
+```
 
+===== Default Values Example:
+```c++
+using myparams = tpu_params_both<tpu::xmx16, int8_t, int8_t, int>;
+// use this to construct the ranges on the host side
+size_t NDRangeM = M / myparams::M;
+size_t NDRangeN = N / myparams::N;
+//if M, N, K do not multiply the default sizes, padding has to be done
+// device code: the matrices are constructed using the default dimensions
+myparams::joint_matrix_a<sub_group, layout::row_major> sub_a;
+myparams::joint_matrix_b<sub_group, layout::row_major> sub_b;
+myparams::joint_matrix_accumulator<sub_group> sub_c;
+
+```
+==== Runtime Query
+This provides a more general query interface with information about
+sizes, types,  and scopes that are supported by a specific matrix
+implementation. This is needed to avoid padding by the user, for
+tuning, and efficient code generation if used by a library. The
+general query returns an array of `combinations` of `combination`
+type. Each combination includes the sizes and the types for the
+matrices A, B, and accumulator. Note that for each matrix hardware,
+the query returns `max_msize, max_nsize, max_ksize` or `msize, nsize,
+ksize` exclusively, depending on whether the implementation supports a
+continuous or discrete number of sizes. For example, the Intel AMX
+implementation supports a continuous number of sizes, so the `max_*`
+variant is applied and only the maximum number is returned. The Intel
+XMX implementation, on the other hand, supports a discrete list of
+numbers so the `msize, nsize, ksize` variant is applied.
+
+The table below provides a description for each of the device matrix
+desciptors that can be queried using `get_info` API.
+
+[frame="none",options="header"]
+|======================
+| Device descriptors | Return type| Description
+|`ext::oneapi::experimental::info::device::matrix::numtiles`| `uint32_t`
+|indicates number of tiles in Intel AMX (does not apply to Intel XMX)
+|`ext::oneapi::experimental::info::device::matrix::scopes`
+|`std::vector<scope_t>`
+|indicates the memory and execution scopes supported by the matrix
+implementation
+|`ext::oneapi::experimental::info::device::matrix::num_scopes`|`uint32_t`
+|indicates number of scopes supported by the matrix implementation
+|`ext::oneapi::experimental::info::device::matrix::combinations` |
+`std::vector<combination>`| tells the set of supported matrix sizes
+and types
+|`ext::oneapi::experimental::info::device::matrix::num_combinations`|`uint32_t`
+|indicates number of combinations supported by the matrix
+implementation which corresponds to the size of the `combinations` array
+|`combination` 
```

Comment:
This is confusing because it mixes the info descriptors together in the same table as the descriptions of the struct members.  Create a new table for the struct members with just two columns "member variable" and "description".  The type is documented in the synopsis, so it doesn't need to be repeated in the table.